bouncer/sentry.py

```diff
@@ -0,0 +1,31 @@
+"""Report exceptions to Sentry/Raven."""
+import raven
+
+
+def raven_tween_factory(handler, _):
+    """Return a tween that reports uncaught exceptions to Sentry."""
+    def raven_tween(request):
+        """Report uncaught exceptions to Sentry."""
+        try:
+            return handler(request)
+        except:
+            request.raven.captureException()
+            raise
+
+    return raven_tween
+
+
+def raven_client(request):
+    """Return the raven.Client object."""
+    request.add_finished_callback(
```

Comment:
I need the request to create the client. The request was being used to access the Pyramid config registry to pass settings to the client. That's no longer necessary, but I'm now using the request to add context (requested URL etc) to the client. I _could_ have a `NewRequest` subscriber add that context instead of doing it in this `raven_client()` function, then I could create the client once at module scope and use the `context.clear()` finished callback. But that spreads code about initialising and configuring the raven client across three different places. Worth the trouble?

I think maybe just creating and configuring new client for each request (and getting rid of the context.clear() as it's unnecessary) is simpler, and expect it doesn't add much workload.